auction-server/src/server.rs

```diff
@@ -140,60 +147,97 @@ pub fn setup_metrics_recorder() -> anyhow::Result<PrometheusHandle> {
-                    .await?;
-            let weth =
-                get_weth_address(chain_config.adapter_factory_contract, provider.clone()).await?;
-            let adapter_bytecode_hash =
-                get_adapter_bytecode_hash(chain_config.adapter_factory_contract, provider.clone())
-                    .await?;
+                        let express_relay_contract = get_express_relay_contract(
+                            chain_config.express_relay_contract,
+                            provider.clone(),
+                            wallet.clone(),
+                            chain_config.legacy_tx,
+                            id,
+                        );
+                        let permit2 = get_permit2_address(
+                            chain_config.adapter_factory_contract,
+                            provider.clone(),
+                        )
+                        .await?;
+                        let weth = get_weth_address(
+                            chain_config.adapter_factory_contract,
+                            provider.clone(),
+                        )
+                        .await?;
+                        let adapter_bytecode_hash = get_adapter_bytecode_hash(
+                            chain_config.adapter_factory_contract,
+                            provider.clone(),
+                        )
+                        .await?;
 
-            Ok((
-                chain_id.clone(),
-                ChainStore {
-                    chain_id_num: id,
-                    provider,
-                    network_id: id,
-                    token_spoof_info: Default::default(),
-                    config: chain_config.clone(),
-                    permit2,
-                    weth,
-                    adapter_bytecode_hash,
-                    express_relay_contract: Arc::new(express_relay_contract),
-                    block_gas_limit: block.gas_limit,
-                },
-            ))
-        }
-    }))
+                        Ok((
+                            chain_id.clone(),
+                            ChainStore {
+                                chain_id_num: id,
+                                provider,
+                                network_id: id,
+                                token_spoof_info: Default::default(),
+                                config: chain_config.clone(),
+                                permit2,
+                                weth,
+                                adapter_bytecode_hash,
+                                express_relay_contract: Arc::new(express_relay_contract),
+                                block_gas_limit: block.gas_limit,
+                            },
+                        ))
+                    })
+                }
+            }),
+    )
     .await
     .into_iter()
     .collect()
 }
 
+pub fn load_express_relay_idl() -> anyhow::Result<Idl> {
+    let idl = fs::read("../express_relay/target/idl/express_relay.json")?;
```

Comment:
I think this is a bad idea, similar to what we have on the evm side, we should include this on compile time and have proper type checking.